### ISSUE TITLE
AoT runtime error for enum VelopackAssetType

### ISSUE THREAD

**Issue by DVaughan** [velopack/velopack#122]
Firstly, thanks for your work on this project. It's very cool.

I've been attempting to make Velopack work in an Avalonia app that uses AoT. There were a few types that were being trimmed away, but I've prevented that by referring to the types and using the `DynamicallyAccessedMembers`. I have full trimming enabled in the .NET 8 application.
However, I stuck now with an enum deserialization issue. When I attempt to query for updates:
```cs
var updateInfo = await updateManager.CheckForUpdatesAsync();
```
I receive:

> Error checking for update.System.NotSupportedException: 'System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1[Velopack.VelopackAssetType]' is missing native code or metadata. This can happen for code that is not compatible with trimming or AOT. Inspect and fix trimming and AOT related warnings that were generated when the app was published. For more information see https://aka.ms/nativeaot-compatibility at System.Reflection.Runtime.General.TypeUnifier.WithVerifiedTypeHandle(RuntimeConstructedGenericTypeInfo, RuntimeTypeInfo[]) + 0x75

It looks like [this](https://stackoverflow.com/questions/77747462/how-to-enable-jsonstringenumconverter-for-all-enums-in-net-aot) might be related.
I've been playing around with the JsonSerializerContext in my own project, but to no avail. I expect it does effect the Velopack serialization, though I may be wrong.

```cs
	[JsonSourceGenerationOptions(UseStringEnumConverter = true)]
	[JsonSerializable(typeof(Velopack.VelopackAssetType))]
	public partial class MyJsonSerializerContext : JsonSerializerContext { }
```

**Comment by DVaughan** [comment]
Link to PR: https://github.com/velopack/velopack/pull/123

**Comment by caesay** [comment]
Thanks for your PR! To support AOT I think it might be better to do a little refactoring and switch System.Text.Json over to using source generators rather than reflection. I think there are 1-2 other places in Velopack that use reflection that could also be eliminated. We probably need to add AOT annotations to prevent types from being trimmed.

What types were being trimmed that you had to fix with DynamicallyAccessedMembers?

**Comment by DVaughan** [comment]
The type that I had issues with initially was VelopackAssetFeed. Trimming was removing parameterless constructors, which was breaking the deserialization. I ended up using `System.Diagnostics.CodeAnalysis.DynamicDependencyAttribute` to prevent trimming of these feed related types: UpdateInfo, VelopackAsset, VelopackAssetType, VelopackAssetFeed. The `VelopackAssetType` enum still didn't work, so that's why I added the generic converter for it, which fixed it.

**Comment by DVaughan** [comment]
I just encountered another serialization issue with `DateTime?`.
> Error checking for update.System.NotSupportedException: 'System.Text.Json.Serialization.Metadata.JsonPropertyInfo`1[System.Nullable`1[System.DateTime]]' is missing native code or metadata. This can happen for code that is not compatible with trimming or AOT. Inspect and fix trimming and AOT related warnings that were generated when the app was published. For more information see https://aka.ms/nativeaot-compatibility at System.Reflection.Runtime.General.TypeUnifier.WithVerifiedTypeHandle(RuntimeConstructedGenericTypeInfo, RuntimeTypeInfo[]) + 0x75

**Comment by caesay** [comment]
Fixed in f32d6f88ff9d2c4a02311085d9db358ccf1acb59
